docs/paper/dev/misc/nms.mdx

```diff
@@ -0,0 +1,66 @@
+---
+slug: /dev/nms
+---
+
+# NMS
+
+NMS is an abbreviation for `net.minecraft.server`. It is the package that contains Mojang's classes that are used by the
+server. This is internal code which often changes between versions. For this reason, plugins that use NMS code are
+often version specific or require different code for different versions. It is often used to refer to CraftBukkit code
+as well.
+
+## Using NMS and CraftBukkit
+
+In order to use NMS and CraftBukkit code, you may either add the dependencies to your plugin or use reflection. Depending
+on the server dependencies is often the easiest way to use NMS and CraftBukkit code. However, reflection often allows
+some code to be used across multiple versions.
+
+### Depending on NMS and CraftBukkit
+
+To access Paper internals, you can use the [paperweight-userdev](https://github.com/PaperMC/paperweight-test-plugin)
+repository. This will allow you to decompile the server and access the NMS and CraftBukkit code. You can
```

Comment:
```suggestion
To access Paper internals, you can use the [paperweight-userdev](https://github.com/PaperMC/paperweight-test-plugin)
gradle plugin. This will allow you to decompile the server and access the NMS and CraftBukkit code. You can
```